code/_core/obj/item/tempering/tempering_quality.dm

```diff
@@ -16,38 +16,43 @@
 
 /obj/item/tempering/quality/can_temper(var/mob/activator,var/obj/item/I)
 
-	if(I.quality <= -1)
+	if(!I.can_adjust_quality())
 		activator.to_chat(span("warning","\The [I.name] cannot be improved!"))
 		return FALSE
 
-	if(I.quality >= maximum)
-		activator.to_chat(span("warning","\The [I.name] cannot be improved any further!"))
-		return FALSE
-
 	return ..()
 
 /obj/item/tempering/quality/get_examine_list(var/mob/examiner)
 	. = ..()
-	. += span("notice","Increases the quality of [item_type_name] by [increase]%, up to [maximum]%, with a minimum of [minimum]%.")
+	. += span("notice","Increases the quality of [item_type_name] by [increase]%, with a minimum of [minimum]%.")
 
-/obj/item/tempering/quality/on_temper(var/mob/activator,var/obj/item/I)
-	I.quality = clamp(I.quality + increase,minimum,maximum)
-	. = ..()
+/obj/item/tempering/quality/on_temper(var/mob/activator, var/obj/item/I)
+	I.adjust_quality((I.quality + increase) > minimum ? increase : minimum - I.quality)
+	return ..()
 
 /obj/item/tempering/quality/general
-	name = "advanced repair kit"
+	name = "advanced repair kitaaaa"
 	desc = "Your gear is your life. Take care of it!"
 	desc_extended = "A collection of scrap, tools, and generally useful materials that can help you maintain the condition any sort of item."
 	icon_state = "repair_kit"
 
 	increase = 25
 	minimum = 50
-	maximum = 100
 
 	temper_whitelist = /obj/item
 
 	value = 400
 
+/obj/item/tempering/quality/general/can_temper(var/mob/activator,var/obj/item/I)
+	if(I.quality >= 100) // We exist only to repair
+		return FALSE
+
+	return ..()
+
+/obj/item/tempering/quality/general/on_temper(var/mob/activator,var/obj/item/I)
+	. = ..()
+	I.quality = min(I.quality, 100)
+
```

Comment:
I don't think there is any reason to include this code since the general tempering stuff handles caps.